docs/apache-airflow/public-airflow-interface.rst

```diff
@@ -0,0 +1,110 @@
+* `Listeners <listeners>`_ - all of which are considered Public Interfaces.
+* `Plugins <plugins>`_, which allow developers to extend internal Airflow capabilities - add new UI
+  pages, custom `TimeTables <concepts/timetable>`_, `Extra Links <howto/define_extra_link>`_,
+
+What is not part of the Public Interface of Apache Airflow?
+===========================================================
+
+Everything not mentioned in this document should be considered as non-Public Interface.
+
+The users however, might have some assumptions about different parts of Airflow, thinking that
+they can rely on them, so here we try to clarify and want to be explicit that they are not part of the
+Public Interface:
+
+* `Database structure <database-erd-ref>`_ is considered to be an internal implementation
+  detail and you should not assume the structure is going to be maintained in
+  backwards-compatible way.
+
+* `Web UI <ui>`_ is considered to be continuously adapting and evolving for Apache Airflow and
+  you should not assume that the UI will remain as it is in backwards-compatible way. Views and screens
+  might be updated and removed in major releases without impacting backwards-compatibility.
+
+* `Python API <python-api-ref>`_ (except the explicitly mentioned classes below), are considered an
+  internal implementation details and you should not assume they will be maintained
+  in backwards-compatible way.
+
+Which classes and packages are part of the Public Interface of Apache Airflow?
+==============================================================================
+
+The Public Interface of Airflow consists of a number of different classes and packages that provide access
+to the core features and functionality of the system.
+
+The classes and packages that may be considered as the Public Interface include:
+
+* The :class:`~airflow.models.dag.DAG`, which provides a way to define and manage DAGs in Airflow.
+* The :class:`~airflow.models.baseoperator.BaseOperator`, which provides a way write custom operators.
+* The :class:`~airflow.hooks.base.BaseHook`, which provides a way write custom hooks.
+* The :class:`~airflow.decorators.base.TaskDecorator`, which provides a way write custom decorators.
+* The :class:`~airflow.models.connection.Connection`, which provides access to external service credentials and configuration.
+* The :class:`~airflow.models.variable.Variable`, which provides access to Airflow configuration variables.
+* The :class:`~airflow.models.xcom.XCom` classes that are used to access to inter-task communication data.
+* The :class:`~airflow.secrets.BaseSecretsBackend` classes that are used to define custom secret managers.
+* The :class:`~airflow.plugins_manager.AirflowPlugin` class that is used to define custom plugins.
+
+.. versionadded:: 2.5
+
+* The :class:`~airflow.listeners.listener` classes that are used to define your own listeners for DAG/Task events
+
+.. versionadded:: 2.6
```

Comment:
These version added don't line up with anything.

Reply:
Restructured the docs a  bit to let them contain content.